### PR TITLE
Skip haystack index creation if aldryn-search is not installed

### LINKED ISSUE #584: Do not import aldryn_search outside PostIndex in search_indexes.py
### Description

We're not using `aldryn_search` at all, but we're using `django-haystack` in order to index elements of our websites. We had installed `djangocms_blog`.

Haystack allows to exclude Indexes with its [EXCLUDED_INDEXES](https://django-haystack.readthedocs.io/en/master/settings.html?highlight=EXCLUDED_INDEXES#haystack-connections) paremeter, but it *only* excludes the content of the Index class.

The problem here is that you include `aldryn_search` submodules *outside* the PostIndex, and so when I prevent your `search_indexes.py` from being used by haystack using `"EXCLUDED_INDEXES": ["djangocms_blog.search_indexes.PostIndex"],`, it always try to import `aldryn_search`, even without using your `PostIndex` at all.

https://github.com/nephila/djangocms-blog/blob/99ab6a1be8a930972d9712321a17ac67a2354bab/djangocms_blog/search_indexes.py#L1-L2

### Steps to reproduce

* Do *not* install `aldryn_search`.
* Try to index something on the project using `django-haystack`, but not djangocms_blog (thanks to `"EXCLUDED_INDEXES": ["djangocms_blog.search_indexes.PostIndex"],`).
* Exception:
   ```bash
   File "[...]djangocms_blog/search_indexes.py", line 4, in <module>
      from aldryn_search.helpers import get_plugin_index_data
   ModuleNotFoundError: No module named 'aldryn_search'
   ```

### Expected solution

Move lines `aldryn_search` imports in `search_indexes.py` inside the class:
https://github.com/nephila/djangocms-blog/blob/99ab6a1be8a930972d9712321a17ac67a2354bab/djangocms_blog/search_indexes.py#L11

### PR DESCRIPTION
**Pull request by yakky**
# Description

If haystack indexes are used, but not aldryn-search, indexing fails due to import of aldryn-search at module level in `djangocms_blog.search_indexes`

By globally catching the import error exception we can skip the error and one can define their own `PostIndex` without aldryn-search

## References

Fix #584 

# Checklist

* [x] I have read the [contribution guide](https://djangocms-blog.readthedocs.io/en/latest/contributing.html)
* [x] Code lint checked via `inv lint`
* [x] ``changes`` file included (see [docs](https://djangocms-blog.readthedocs.io/en/latest/contributing.html#pull-request-guidelines))
* [x] Usage documentation added in case of new features
* [x] Tests added

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -49,6 +49,8 @@ If you want to enable haystack support, in addition to the above:
 * add ``aldryn_search`` to ``INSTALLED_APPS``
 * configure haystack according to `aldryn-search docs <https://github.com/aldryn/aldryn-search#usage>`_
   and `haystack docs <http://django-haystack.readthedocs.io/en/stable/>`_.
+* if not using ``aldryn_search``, you can define your own index by skipping ``aldryn_search`` installation and writing
```
Comment (corentinbettiol) [inline docs/installation.rst:52]: Maybe replace `your index` by `your own search_indexes.py` in order to give potential users a hint of what to add ?

Comment (yakky) [reply docs/installation.rst]: thanks for the suggestion, I reworded the documentation according to this comment

**Comment by yakky** [comment]
> That's the biggest try except pass I've ever seen, but it seems to be the best way to check is aldryn_search is used.

As we use an aldryn-search base class there is no way around a by try / except or a big if clause, and the formed looked a bit better to me :shrug:

**Comment by corentinbettiol** [comment]
Yes I think that's the best option we have in order to allow custom search engines based on haystack to work flawlessly with djangocms-blog.

Thanks for this PR :)

**Pull request merged**